### PR TITLE
OSSM-3692 Refactor JWT Authorization (T22)

### PR DESCRIPTION
**Pull request by pbajjuri20**

### CODE REVIEW

File: pkg/tests/tasks/security/authorization/deny_test.go [hunk 1]
```diff
@@ -48,7 +48,7 @@ func TestAuthorizationDenyAllow(t *testing.T) {
 			oc.ApplyString(t, ns, DenyGETPolicy)
 
 			t.LogStep("Verify that GET request is denied")
-			assertRequestDenied(t, ns, httpbinRequest("GET", "/get"))
+			assertRequestDenied(t, ns, httpbinRequest("GET", "/get"), "403", "Got the expected 403 Forbidden response")
```
Comment (luksa) [inline pkg/tests/tasks/security/authorization/deny_test.go:51]: Passing in the entire success message is not that great. It's better if this function accepts only the httpStatusCode and then formats the success message on its own (IMHO the message doesn't really need to say "Forbidden"... a simple `fmt.Sprintf("Got the expected %s response code", httpStatusCode)` is fine).

File: pkg/tests/tasks/security/authorization/jwt_test.go [hunk 2]
```diff
@@ -15,160 +15,131 @@
...
+				oc.DeleteFromString(t, ns, JWTRequireRule)
+			})
+			oc.ApplyString(t, ns, JWTRequireRule)
+
+			t.LogStep("Verify that a request with a valid JWT is allowed")
+			assertRequestAccepted(t, ns, httpbinRequest("GET", "/headers", "Authorization: Bearer %s", token))
+
+			t.LogStep("Verify request without a JWT is denied")
+			assertRequestDenied(t, ns, httpbinRequest("GET", "/headers"), "403", "Got the expected 403 Forbidden response")
+
```
Comment (luksa) [inline pkg/tests/tasks/security/authorization/jwt_test.go:75]: Unnecessary empty line.

File: pkg/tests/tasks/security/authorization/jwt_test.go [hunk 3]
```diff
@@ -15,160 +15,131 @@
...
+				oc.DeleteFromString(t, ns, JWTGroupClaimRule)
+			})
+			oc.ApplyString(t, ns, JWTGroupClaimRule)
+
+			t.LogStep("Verify that a request with the JWT that includes group1 in the groups claim is allowed")
+			assertRequestAccepted(t, ns, httpbinRequest("GET", "/headers", "Authorization: Bearer %s", tokenGroup))
+
+			t.LogStep("Verify that a request with a JWT, which does not have the groups claim is rejected")
+			assertRequestDenied(t, ns, httpbinRequest("GET", "/headers", "Authorization: Bearer %s", token), "403", "Got the expected 403 Forbidden response")
+
```
Comment (luksa) [inline pkg/tests/tasks/security/authorization/jwt_test.go:89]: Unnecessary empty line.

**Review by luksa — CHANGES_REQUESTED** [review]

File: pkg/tests/tasks/security/authorization/common.go [hunk 4]
```diff
@@ -44,15 +44,15 @@ func assertRequestAccepted(t test.TestHelper, ns string, curlCommand string) {
 	})
 }
 
-func assertRequestDenied(t test.TestHelper, ns string, curlCommand string) {
+func assertRequestDenied(t test.TestHelper, ns string, curlCommand string, expectedStatusCode string, expectedStatus string) {
```
Comment (luksa) [inline pkg/tests/tasks/security/authorization/common.go:47]: Unused parameter `expectedStatus`

Comment (luksa) [reply pkg/tests/tasks/security/authorization/common.go:47]: Just remove it.

**Comment by pbajjuri20** [comment]
@luksa: I have fixed the comments and pushed the latest commit.

File: pkg/tests/tasks/security/authorization/common.go (hunk 4 above)
Comment (pbajjuri20) [reply pkg/tests/tasks/security/authorization/common.go]: @luksa: removed the unnecessary parameter and pushed the latest change.

**Review by luksa — APPROVED** [review]

**Pull request merged**